protocol/v2/types/ssvshare_test.go

```diff
@@ -105,3 +106,82 @@ func TestValidCommitteeSize(t *testing.T) {
 		})
 	}
 }
+
+func TestSSVShare_IsActive(t *testing.T) {
+	currentEpoch := phase0.Epoch(100) // Example current epoch for testing
+	tt := []struct {
+		Name     string
+		Share    *SSVShare
+		Epoch    phase0.Epoch
+		Expected bool
+	}{
+		{
+			Name: "No BeaconMetadata",
+			Share: &SSVShare{
+				Metadata: Metadata{
+					BeaconMetadata: nil,
+				},
+			},
+			Epoch:    currentEpoch,
+			Expected: false,
+		},
+		{
+			Name: "Is Attesting",
+			Share: &SSVShare{
+				Metadata: Metadata{
+					BeaconMetadata: &beaconprotocol.ValidatorMetadata{
+						Status: eth2apiv1.ValidatorStateActiveOngoing,
+					},
+				},
+			},
+			Epoch:    currentEpoch,
+			Expected: true,
+		},
+		{
+			Name: "Pending Queued with Future Activation Epoch",
+			Share: &SSVShare{
+				Metadata: Metadata{
+					BeaconMetadata: &beaconprotocol.ValidatorMetadata{
+						Status:          eth2apiv1.ValidatorStatePendingQueued,
+						ActivationEpoch: currentEpoch + 1,
+					},
+				},
+			},
+			Epoch:    currentEpoch,
+			Expected: false,
+		},
+		{
+			Name: "Pending Queued with Current Epoch as Activation Epoch",
+			Share: &SSVShare{
+				Metadata: Metadata{
+					BeaconMetadata: &beaconprotocol.ValidatorMetadata{
+						Status:          eth2apiv1.ValidatorStatePendingQueued,
+						ActivationEpoch: currentEpoch,
+					},
+				},
+			},
+			Epoch:    currentEpoch,
+			Expected: true,
+		},
+		{
+			Name: "Pending Queued with Past Activation Epoch",
+			Share: &SSVShare{
+				Metadata: Metadata{
+					BeaconMetadata: &beaconprotocol.ValidatorMetadata{
+						Status:          eth2apiv1.ValidatorStatePendingQueued,
+						ActivationEpoch: currentEpoch - 1,
+					},
+				},
+			},
+			Epoch:    currentEpoch,
+			Expected: true,
+		},
+	}
+
+	for _, tc := range tt {
```

Comment:
Since we are not using Go 1.22 yet, let's add this line

```suggestion
	for _, tc := range tt {
		tc := tc
```

Reply:
good catch @nkryuchkov